### DISCUSSION TITLE
Cask: Plex Media Server Application Support Directory

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [Homebrew/discussions#4781] [Casks] [answered]
### Output of `brew config`

```shell
I don't know... running in GitHub actions, and irrelevant to my question
```

### Output of `brew doctor`

```shell
I don't know... running in GitHub actions, and irrelevant to my question
```

### Description of issue

What is the `Application Support` directory of the Plex Media Server cask?

According to Plex (https://support.plex.tv/articles/202915258-where-is-the-plex-media-server-data-directory-located/#toc-1) it should be `~/Library/Application Support/Plex Media Server/`

But this path appears to be trashed?

https://github.com/Homebrew/homebrew-cask/blob/38fbb71fd0ea851b2feeb574dc3935662610301d/Casks/p/plex-media-server.rb#L28

### Relevant casks

plex-media-server

**Comment by cho-m** [comment]
> What is the `Application Support` directory of the Plex Media Server cask?

It should match the location Plex claims.

---

> But this path appears to be trashed?

Zap is only used as extra option when uninstalling (e.g. `brew uninstall --zap plex-media-server`) for users who want to remove all extra cruft (preferences, cached data, etc) related to cask. Those files/directories remain while installed and using a standard uninstall (e.g. `brew uninstall plex-media-server`)

---

EDIT: See https://docs.brew.sh/Cask-Cookbook#zap-purpose for more details

**Reply by ReenigneArcher** [reply]
Interesting, but the directory does not exist.

```bash
brew install --cask plex-media-server

# starting with pms 1.29.2 servers must be claimed... disable that
# https://forums.plex.tv/t/new-server-claiming-requirement-for-macos/816337
defaults write com.plexapp.plexmediaserver enableLocalSecurity -bool FALSE

open "/Applications/Plex Media Server.app"
ls -a "~/Library/Application Support/Plex Media Server"
```

```txt
ls: ~/Library/Application Support/Plex Media Server: No such file or directory
```

**Reply by cho-m** [reply]
At least for above output, don't put `~` inside double quotes. It will be interpreted as the name of a directory and not expanded.

Anyways, the app from Cask is identical to official app. The installation does the same thing as manual install where the `.app` is copied into `/Applications` directory. Homebrew can't modify the upstream's binary so the settings they set up will be the same.

**Reply by ReenigneArcher** [reply]
Interesting about the `~`. I did not know that.

```bash
cd ~/Library
cd "./Application Support"
ls -a .
```

There's no `Plex Media Server` directory listed.

Could this possibly be due to how I'm opening plex? Using `open "/Applications/Plex Media Server.app"`

Is there another preferred way to open plex from terminal?

**Reply by cho-m** [reply]
It could be that there is a GUI prompt either from app or on Mac side (e.g. Apple quarantine prompt). Not sure if possible, but may want to see if any way to access machine with display to see.

Apple's quarantine prompt is a common situation so you could try disabling it to see if it helps (`brew install --no-quarantine ...`) assuming you are fine with that (ref: https://docs.brew.sh/FAQ#why-cant-i-open-a-mac-app-from-an-unidentified-developer)

**Reply by ReenigneArcher** [reply]
I appreciate the help so far. Adding `--no-quarantine` seems to produce the same results.

I can run the plex server with `open "/Applications/Plex Media Server.app"`. I think you are saying with the quarantine prompt that I wouldn't even be able to run the server at all?

Unfortunately I don't have access to any physical macOS machine.

**Reply by cho-m** [reply, marked as answer]
> I think you are saying with the quarantine prompt that I wouldn't even be able to run the server at all?

Yeah, if it was quarantine/Gatekeeper-related, app would be blocked and probably wouldn't show up as running in `ps` output.

---

You could try doing a full disk search using something like `find` to see if directory exists elsewhere.

No idea otherwise. May want to see if can get help from Plex side.

**Reply by ReenigneArcher** [reply]
Alright. I did do a search on the disk using `find / -type d -name "Plex Media Server" 2>/dev/null
`

That took about 35 minutes to finish but found the following.

```txt
/System/Volumes/Data/Users/runner/Library/Application Support/Plex Media Server
/System/Volumes/Data/Users/runner/Library/Logs/Plex Media Server
/Users/runner/Library/Application Support/Plex Media Server
/Users/runner/Library/Logs/Plex Media Server
```

I guess the ones are under `/System` are symbolic links or something.

Anyway, based on this it appears the directories are created when plex is first started, but it takes a bit of time. I adjusted my CI script to install the plugin before starting the Plex server. For Windows and Ubuntu I was able to do it with the server already running. Not sure why that didn't work on macOS, but doing it as a prep step appears to be working.

I appreciate all the help... and props to the Homebrew project. It has made life pretty easy for CI pipelines on macOS.
